app/src/main/java/com/sdpteam/connectout/qr_code/QRcodeActivity.java

```diff
@@ -0,0 +1,57 @@
+package com.sdpteam.connectout.qr_code;
+
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.widget.Button;
+
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.appcompat.app.AlertDialog;
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.journeyapps.barcodescanner.ScanContract;
+import com.journeyapps.barcodescanner.ScanIntentResult;
+import com.journeyapps.barcodescanner.ScanOptions;
+import com.sdpteam.connectout.R;
+
+public class QRcodeActivity extends AppCompatActivity {
+
+    private Button btn_scan;
+    private ActivityResultLauncher<ScanOptions> barLauncher;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_qrcode);
+
+        btn_scan = findViewById(R.id.btn_scan);
+        btn_scan.setOnClickListener(v -> scanCode());
+
+        barLauncher = registerForActivityResult(new ScanContract(), result -> {
+            handleScanResult(result.getContents());
+        });
+    }
+
+    private void scanCode() {
+        ScanOptions options = new ScanOptions();
+        options.setPrompt("Volum up to flash on");
+        options.setBeepEnabled(true);
+        options.setOrientationLocked(true);
+        options.setCaptureActivity(CaptureAct.class);
+
+        barLauncher.launch(options);
+    }
+
+    void handleScanResult(String resultText) {
+        if (resultText != null) {
+            showDialog(resultText);
```

Comment:
At this point you should parse the resultText in order to create an intent to open the event or profile activity